src/Filter/EmptyFieldFilter.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\DoctrineORMAdminBundle\Filter;
+
+use Sonata\AdminBundle\Datagrid\ProxyQueryInterface;
+use Sonata\AdminBundle\Form\Type\Filter\DefaultType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+
+final class EmptyFieldFilter extends Filter
+{
+    /**
+     * Field is null.
+     */
+    public const TYPE_EMPTY = 1;
+
+    /**
+     *  Field is not null.
+     */
+    public const TYPE_NOT_EMPTY = 2;
+
+    /**
+     * @param string            $alias
+     * @param string            $field
+     * @param array|string|null $data
+     */
+    public function filter(ProxyQueryInterface $queryBuilder, $alias, $field, $data): void
+    {
+        if (null === $data || !\is_array($data) || !\array_key_exists('value', $data)) {
+            return;
+        }
+
+        if ((int) $data['value'] === static::TYPE_NOT_EMPTY) {
+            $queryBuilder->andWhere(sprintf('%s.%s IS null', $alias, $field));
+        } elseif ((int) $data['value'] === static::TYPE_EMPTY) {
+            $queryBuilder->andWhere(sprintf('%s.%s IS NOT null', $alias, $field));
+        }
+
+        $this->active = null !== $data['value'];
+    }
+
+    public function getDefaultOptions()
+    {
+        return [];
+    }
+
+    public function getFieldType()
+    {
+        return $this->getOption('field_type', ChoiceType::class);
+    }
+
+    public function getFieldOptions()
+    {
+        return $this->getOption('choices', [
+            'choices' => [
+                'label_type_empty' => static::TYPE_EMPTY,
+                'label_type_not_empty' => static::TYPE_NOT_EMPTY,
+            ],
+        ]);
+    }
+
+    public function getRenderSettings()
+    {
+        return [DefaultType::class, [
```

Comment:
I think you should create a SonataAdminBundle/Form/Type.
For example the `NumberFilter` use a `NumberType`.

Since there is multiple ORM Repository, this is a way to use same labels, constants, configuration, ...

Reply:
An "EmptyOperatorType" like this https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Form/Type/Operator/ContainsOperatorType.php in the SonataAdminBundle would be better.

This allow to be used by others PersistenceBundle.